apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/RetryOptions.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+namespace Google.Cloud.Storage.V1;
+
+/// <summary>
+/// Options used to indicate the retry configuration for an API request in case of failure. This class is immutable.
+/// These determine whether and how the request should be retried.
+/// </summary>
+public sealed class RetryOptions
+{
+    /// <summary>
+    /// Options used by default for idempotent operations upon failure.
+    /// It has all retry timings reset to defaults and retry predicate based on default error codes.
+    /// </summary>
+    public static RetryOptions IdempotentRetryOptions { get; } = new RetryOptions(RetryTiming.Default, RetryPredicate.RetriableIdempotentErrors);
+
+    /// <summary>
+    /// This configuration ensures that the API request is never retried on failure.
+    /// </summary>
+    public static RetryOptions Never { get; } = new RetryOptions(RetryTiming.Default, RetryPredicate.Never);
+
+    internal RetryPredicate Predicate { get; }
+    internal RetryTiming Timing { get; }
+
+    internal static RetryOptions MaybeIdempotent(object condition) =>
```

Comment:
I'd add an XML comment for this, as it's not *hugely* clear how `condition` is going to be used.

Reply:
Please fill in the `<returns></returns>` part, or just remove it.